packages/core/src/amazonq/activation.ts

```diff
@@ -17,6 +17,86 @@ import { listCodeWhispererCommandsWalkthrough } from '../codewhisperer/ui/status
 import { Commands } from '../shared/vscode/commands2'
 import { focusAmazonQPanel, focusAmazonQPanelKeybinding } from '../codewhispererChat/commands/registerCommands'
 import { TryChatCodeLensProvider, tryChatCodeLensCommand } from '../codewhispererChat/editor/codelens'
+import { Auth } from '../auth'
+import { customLearnMoreUri } from '../codewhisperer/models/constants'
+import { openUrl } from '../shared/utilities/vsCodeUtils'
+import { AuthUtil } from '../codewhisperer'
+import { ConnectionStateChangeEvent } from '../auth/auth'
+
+/**
+ * Display a notification to user for Log In.
+ *
+ * Authentication Notification is displayed when:
+ * - The user closes the Amazon Q chat panel, and
+ * - The user has not performed any authentication action.
+ *
+ * Error Notification is displayed when:
+ * - The user closes the Amazon Q chat panel, and
+ * - The user attempts an authentication action but is not logged in.
+ *
+ * @param {Event} isAmazonQVisibleEvent - Event indicating the visibility status of the Amazon Q chat.
+ * @param {Event} onDidUpdateConnectionEvent - Event indicating the authentication connection update.
+ */
+async function showAuthNotification(
+    isAmazonQVisibleEvent: vscode.Event<boolean>,
+    onDidUpdateConnectionEvent: vscode.Event<ConnectionStateChangeEvent | undefined>
+) {
+    let isAmazonQVisible = true // Assume Chat is open by default.
+    let notificationDisplayed = false // Auth Notification should be displayed only once.
+    let authConnection: ConnectionStateChangeEvent
+
+    // Updates the visibility state of the Amazon Q chat.
+    const updateVisibility = async (visible: boolean) => {
+        isAmazonQVisible = visible
+        await showNotification()
+    }
+
+    // Updates the source of the connection for Amazon Q sign in.
+    const updateConnection = async (connection: ConnectionStateChangeEvent | undefined) => {
+        if (connection) {
+            authConnection = connection
+            await showNotification()
+        }
+    }
+
+    const disposables: vscode.Disposable[] = [
+        isAmazonQVisibleEvent(updateVisibility),
+        onDidUpdateConnectionEvent(updateConnection),
+    ]
+
+    async function showNotification() {
```

Comment:
Since this only shows the notification on certain conditions we should use `try***` to imply that

```suggestion
    async function tryShowNotification() {
```